### PR TITLE
Integrate Gator with workflow to enable parallel runs

### PR DESCRIPTION
**Pull request by Intuity**
This PR integrates Gator to enable parallel task execution within workflows, this is off by default but can be enabled and controlled with two flags:

 * `bw wf --parallel` - enables the parallel runner
 * `bw wf --parallel --concurrency 4` - sets the number of jobs that may execute in parallel 

Gator only currently supports a simple local scheduler, but will be extended in future to integrate with Slurm/SGE

### CODE REVIEW

File: example/infra/tools/objstore.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
+# Copyright 2024, Peter Birch, mailto:peter@intuity.io
```
Comment (Kotarski) [inline example/infra/tools/objstore.py:1]: We've used
```
# Copyright 2023, Blockwork, github.com/intuity/blockwork
```
elsewhere

File: blockwork/activities/workflow.py [hunk 2]
```diff
@@ -23,3 +31,20 @@ def wf() -> None:
...
+    Loads a serialised transform specification from a provided file path, then
+    resolves the transform class and executes it. This should NOT be called
+    directly but instead as part of a wider workflow.
+    """
+    spec = json.loads(spec_path.read_text(encoding="utf-8"))
+    # Import the relevant transform
+    mod_spec = importlib.import_module(spec["mod"])
+    transform_cls: type[Transform] = getattr(mod_spec, spec["name"])
+    logging.info(f"Running serialised {transform_cls.__name__}")
+    transform_cls._run_serialized(ctx, spec)
```
Comment (Kotarski) [inline blockwork/activities/workflow.py:50]: We will need to add cache store functionality (probably here) if we want parallel to work with caching (doesn't have to be this PR).

We'll probably want to make the non-parallel functionality also run wf_step so we aren't duplicating

File: example/infra/tools/objstore.py (hunk 1 above)
Comment (Intuity) [reply example/infra/tools/objstore.py]: Fixed

File: blockwork/activities/workflow.py (hunk 2 above)
Comment (Intuity) [reply blockwork/activities/workflow.py:55]: Added a big TODO covering these points

**Review by Kotarski — APPROVED** [review]

**Pull request merged**